test/jdk/java/io/BufferedInputStream/TransferToTrusted.java

```diff
@@ -58,38 +59,31 @@ public void write(int b) throws IOException {
         }
     }
 
-    public static void main(String[] args) throws IOException {
+    public static void main(String[] args) throws Exception {
         final int length = 128;
         byte[] buf = new byte[length];
         RND.nextBytes(buf);
-        byte[] dup = Arrays.copyOf(buf, buf.length);
 
-        var bis = new BufferedInputStream(new ByteArrayInputStream(dup));
-        bis.mark(dup.length);
+        var outputStreams = new OutputStream[]{
+                new ByteArrayOutputStream(),
+                new FileOutputStream(File.createTempFile(TransferToTrusted.class.getName(), null)),
+                new PipedOutputStream(new PipedInputStream(length)),
+                new UntrustedOutputStream()
+        };
 
-        var tempFile = File.createTempFile(TransferToTrusted.class.getName(), null);
-        try {
-            var outputStreams = new OutputStream[]{
-                    new ByteArrayOutputStream(),
-                    new FileOutputStream(tempFile),
-                    new PipedOutputStream(new PipedInputStream(length)),
-                    new UntrustedOutputStream()
-            };
-
-            for (var out : outputStreams) {
-                System.err.println("out: " + out.getClass().getName());
+        for (var out : outputStreams) {
+            System.err.println("out: " + out.getClass().getName());
 
+            var bis = new BufferedInputStream(new ByteArrayInputStream(buf.clone()));
+            try (out; bis) {
+                bis.read();//need this to fill the BIS.buf in
                 bis.transferTo(out);
-                bis.reset();
-                try (out) {
-                    if (!Arrays.equals(buf, bis.readAllBytes())) {
-                        throw new RuntimeException("Internal buffer was modified");
-                    }
+                var internalBuffer = bis.getClass().getDeclaredField("buf");
```

Comment:
IMHO you don't need to whitebox here, as the blackbox approach described by me earlier should be sufficient to assert the claim of this PR.